### DISCUSSION TITLE
Getting valueError when running unit test cases

### DISCUSSION THREAD

**Discussion by DineshGon** [fastapi/fastapi#6121] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def test_create_user():
    data = {"firstname": 'tst', 'lastname': 'lst'}
    response = client.post('/user', json=data)
    assert response.status_code == 201
    assert response.json() == data
```

### Description

I am getting the below issue when running the unit test cases
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

-- self =  ProactorEventLoop running=False closed=False debug=False

    def close(self):
        if self.is_running():
            raise RuntimeError("Cannot close a running event loop")
        if self.is_closed():
            return
>       signal.set_wakeup_fd(-1)
E       ValueError: set_wakeup_fd only works in main thread

C:\Program Files\Python38\lib\asyncio\proactor_events.py:679: ValueError

### Operating System

Windows

### FastAPI Version

0.85.0

### Python Version

3.8

**Comment by JarroVGIT** [comment]
Please post all relevant code and stacktrace. In this case: your endpoint function, how you initiate the testclient and the full stacktrace (in code format). Thanks!

**Comment by DineshGon** [comment]
https://github.com/fargonmer/FastAPI . Here is my source code repo

**Comment by DineshGon** [comment]
This is the stack trace when i ran the tests
============================= test session starts =============================
collecting ... collected 1 item

test_vaccineRouter.py::test_create_user FAILED                        [100%]
test_vaccineRouter.py:23 (test_create_user)
def test_create_user():
     data = {"firstname": 'tst', 'lastname': 'lst'}
 . >       response = client.post('/user', json=data)
test_vaccineRouter.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\venv\lib\site-packages\requests\sessions.py:577: in post
    return self.request('POST', url, data=data, json=json, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:473: in request
    return super().request(
..\venv\lib\site-packages\requests\sessions.py:529: in request
    resp = self.send(prep, **send_kwargs)
..\venv\lib\site-packages\requests\sessions.py:645: in send
    r = adapter.send(request, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:267: in send
    raise exc
..\venv\lib\site-packages\starlette\testclient.py:264: in send
    portal.call(self.app, scope, receive, send)
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\starlette\testclient.py:451: in _portal_factory
    yield portal
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\anyio\from_thread.py:502: in start_blocking_portal
    run_future.result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:439: in result
    return self.__get_result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:388: in __get_result
    raise self._exception
C:\Program Files\Python38\lib\concurrent\futures\thread.py:57: in run
    result = self.fn(*self.args, **self.kwargs)
..\venv\lib\site-packages\anyio\_core\_eventloop.py:70: in run
    return asynclib.run(func, *args, **backend_options)
..\venv\lib\site-packages\anyio\_backends\_asyncio.py:292: in run
    return native_run(wrapper(), debug=debug)
C:\Program Files\Python38\lib\asyncio\runners.py:50: in run
    loop.close()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ProactorEventLoop running=False closed=False debug=False>

    def close(self):
        if self.is_running():
            raise RuntimeError("Cannot close a running event loop")
        if self.is_closed():
            return
    
>       signal.set_wakeup_fd(-1)
E       ValueError: set_wakeup_fd only works in main thread

C:\Program Files\Python38\lib\asyncio\proactor_events.py:679: ValueError


============================== 1 failed in 1.08s ==============================

Process finished with exit code 1

**Comment by JarroVGIT** [comment]
I don't see any `def test_create_user()` test function in your test, nor can I find the file `test_vaccineRouter.py`.

**Comment by DineshGon** [comment]
Please find the updated link. I slightly modified the above. 
https://github.com/fargonmer/FastAPI/blob/main/vaccine/tests/testvaccine.py

**Comment by JarroVGIT** [comment]
Sorry to be a drag here, but your stack trace does not match your file. And without it matching there is not really much we can go on. You are pointing to a file called `testvaccine.py`(I found that as well) and your stacks trace is about `test_vaccineRouter.py`.

**Comment by DineshGon** [comment]
Both files are same. in my personal system i used testvaccine.py whereas in my another system i used test_vaccineRouter.py.
Content was same though in both of those files

**Comment by JarroVGIT** [comment]
Except that they are definitely not the same, as you are calling the test function `test_create_user()` which does not exist in the file you linked to.

Anyway, I've cloned your repo that you did link to, and ran the tests, and I cannot reproduce:
```
rootdir: /Users/jarro/Development/fastapi-github-issues/FastAPI/vaccine
plugins: anyio-3.6.1, cov-3.0.0
collected 6 items                                                                                                                             

tests/testvaccine.py ......                                                                                                             [100%]

============================================================== 6 passed in 0.27s ==============================================================
(venv) jarro@MBP-van-Jarro vaccine %

**Comment by DineshGon** [comment]
Below is the msg which i am getting 
(venv) PS C:\Users\Downloads\FastAPI-main\vaccine> pytest
========================================================================================================= test session starts =========================================================================================================
platform win32 -- Python 3.8.0, pytest-7.1.3, pluggy-1.0.0
rootdir: C:\Users\C233696\Downloads\FastAPI-main\vaccine
plugins: anyio-3.6.1, bdd-6.0.1
collected 0 items

======================================================================================================== no tests ran in 0.22s ========================================================================================================
(venv) PS C:\Users\Downloads\FastAPI-main\vaccine>

**Comment by JarroVGIT** [comment]
Try `pytest tests/testvaccine.py`.

anyway, this is completely unrelated to the opening issue. Perhaps close this one, and start a new one with an actual reproducible example (so; the stacktrace **and the code you actually executed to get that stack trace**).

**Comment by DineshGon** [comment]
Will do. Thank you. Even though I run the above command still it fails
(venv) PS C:\Users\Downloads\FastAPI-main\vaccine> pytest tests/testvaccine.py
========================================================================================================= test session starts =========================================================================================================
platform win32 -- Python 3.8.0, pytest-7.1.3, pluggy-1.0.0
rootdir: C:\Users\C233696\Downloads\FastAPI-main\vaccine
plugins: anyio-3.6.1, bdd-6.0.1
collected 6 items

tests\testvaccine.py FFFFFF                                                                                                                                                                                                      [100%]

============================================================================================================== FAILURES =============================================================================================================== 
___________________________________________________________________________________________________ test_retrieve_appointments_byid ___________________________________________________________________________________________________

    def test_retrieve_appointments_byid():
        data = {"startTime": '3:30', 'endTime': '5:40'}
>       response = client.post('/devsecops/vaccine', json.dumps(data))

tests\testvaccine.py:14:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
..\venv\lib\site-packages\requests\sessions.py:635: in post
    return self.request("POST", url, data=data, json=json, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:473: in request
    return super().request(
..\venv\lib\site-packages\requests\sessions.py:587: in request
    resp = self.send(prep, **send_kwargs)
..\venv\lib\site-packages\requests\sessions.py:701: in send
    r = adapter.send(request, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:267: in send
    raise exc
..\venv\lib\site-packages\starlette\testclient.py:264: in send
    portal.call(self.app, scope, receive, send)
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\starlette\testclient.py:451: in _portal_factory
    yield portal
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\anyio\from_thread.py:502: in start_blocking_portal
    run_future.result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:439: in result
    return self.__get_result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:388: in __get_result
    raise self._exception
C:\Program Files\Python38\lib\concurrent\futures\thread.py:57: in run
    result = self.fn(*self.args, **self.kwargs)
..\venv\lib\site-packages\anyio\_core\_eventloop.py:70: in run
    return asynclib.run(func, *args, **backend_options)
..\venv\lib\site-packages\anyio\_backends\_asyncio.py:292: in run
    return native_run(wrapper(), debug=debug)
C:\Program Files\Python38\lib\asyncio\runners.py:50: in run
    loop.close()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ProactorEventLoop running=False closed=False debug=False>

    def close(self):
        if self.is_running():
            raise RuntimeError("Cannot close a running event loop")
        if self.is_closed():
            return

>       signal.set_wakeup_fd(-1)
E       ValueError: set_wakeup_fd only works in main thread

C:\Program Files\Python38\lib\asyncio\proactor_events.py:679: ValueError
_________________________________________________________________________________________________________ test_create_vaccine _________________________________________________________________________________________________________ 

    def test_create_vaccine():
        data = {"startTime": '3:30', 'endTime': '5:40'}
>       response = client.post('/devsecops/vaccine', json.dumps(data))

tests\testvaccine.py:24:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
..\venv\lib\site-packages\requests\sessions.py:635: in post
    return self.request("POST", url, data=data, json=json, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:473: in request
    return super().request(
..\venv\lib\site-packages\requests\sessions.py:587: in request
    resp = self.send(prep, **send_kwargs)
..\venv\lib\site-packages\requests\sessions.py:701: in send
    r = adapter.send(request, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:267: in send
    raise exc
..\venv\lib\site-packages\starlette\testclient.py:264: in send
    portal.call(self.app, scope, receive, send)
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\starlette\testclient.py:451: in _portal_factory
    yield portal
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\anyio\from_thread.py:502: in start_blocking_portal
    run_future.result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:439: in result
    return self.__get_result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:388: in __get_result
    raise self._exception
C:\Program Files\Python38\lib\concurrent\futures\thread.py:57: in run
    result = self.fn(*self.args, **self.kwargs)
..\venv\lib\site-packages\anyio\_core\_eventloop.py:70: in run
    return asynclib.run(func, *args, **backend_options)
..\venv\lib\site-packages\anyio\_backends\_asyncio.py:292: in run
    return native_run(wrapper(), debug=debug)
C:\Program Files\Python38\lib\asyncio\runners.py:50: in run
    loop.close()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ProactorEventLoop running=False closed=False debug=False>

    def close(self):
        if self.is_running():
            raise RuntimeError("Cannot close a running event loop")
        if self.is_closed():
            return

>       signal.set_wakeup_fd(-1)
E       ValueError: set_wakeup_fd only works in main thread

C:\Program Files\Python38\lib\asyncio\proactor_events.py:679: ValueError
___________________________________________________________________________________________________ test_retrieve_all_appointments ____________________________________________________________________________________________________ 

    def test_retrieve_all_appointments():
>       response = client.get('/devsecops/vaccine/all')

tests\testvaccine.py:31:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
..\venv\lib\site-packages\requests\sessions.py:600: in get
    return self.request("GET", url, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:473: in request
    return super().request(
..\venv\lib\site-packages\requests\sessions.py:587: in request
    resp = self.send(prep, **send_kwargs)
..\venv\lib\site-packages\requests\sessions.py:701: in send
    r = adapter.send(request, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:267: in send
    raise exc
..\venv\lib\site-packages\starlette\testclient.py:264: in send
    portal.call(self.app, scope, receive, send)
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\starlette\testclient.py:451: in _portal_factory
    yield portal
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\anyio\from_thread.py:502: in start_blocking_portal
    run_future.result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:432: in result
    return self.__get_result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:388: in __get_result
    raise self._exception
C:\Program Files\Python38\lib\concurrent\futures\thread.py:57: in run
    result = self.fn(*self.args, **self.kwargs)
..\venv\lib\site-packages\anyio\_core\_eventloop.py:70: in run
    return asynclib.run(func, *args, **backend_options)
..\venv\lib\site-packages\anyio\_backends\_asyncio.py:292: in run
    return native_run(wrapper(), debug=debug)
C:\Program Files\Python38\lib\asyncio\runners.py:50: in run
    loop.close()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ProactorEventLoop running=False closed=False debug=False>

    def close(self):
        if self.is_running():
            raise RuntimeError("Cannot close a running event loop")
        if self.is_closed():
            return

>       signal.set_wakeup_fd(-1)
E       ValueError: set_wakeup_fd only works in main thread

C:\Program Files\Python38\lib\asyncio\proactor_events.py:679: ValueError
_______________________________________________________________________________________________________ test_update_appointment _______________________________________________________________________________________________________ 

    def test_update_appointment():
        data = {"startTime": '1:10', 'endTime': '2:20'}
>       response = client.put('devsecops/vaccine/1', json.dumps(data))

tests\testvaccine.py:37:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
..\venv\lib\site-packages\requests\sessions.py:647: in put
    return self.request("PUT", url, data=data, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:473: in request
    return super().request(
..\venv\lib\site-packages\requests\sessions.py:587: in request
    resp = self.send(prep, **send_kwargs)
..\venv\lib\site-packages\requests\sessions.py:701: in send
    r = adapter.send(request, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:267: in send
    raise exc
..\venv\lib\site-packages\starlette\testclient.py:264: in send
    portal.call(self.app, scope, receive, send)
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\starlette\testclient.py:451: in _portal_factory
    yield portal
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\anyio\from_thread.py:502: in start_blocking_portal
    run_future.result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:439: in result
    return self.__get_result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:388: in __get_result
    raise self._exception
C:\Program Files\Python38\lib\concurrent\futures\thread.py:57: in run
    result = self.fn(*self.args, **self.kwargs)
..\venv\lib\site-packages\anyio\_core\_eventloop.py:70: in run
    return asynclib.run(func, *args, **backend_options)
..\venv\lib\site-packages\anyio\_backends\_asyncio.py:292: in run
    return native_run(wrapper(), debug=debug)
C:\Program Files\Python38\lib\asyncio\runners.py:50: in run
    loop.close()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ProactorEventLoop running=False closed=False debug=False>

    def close(self):
        if self.is_running():
            raise RuntimeError("Cannot close a running event loop")
        if self.is_closed():
            return

>       signal.set_wakeup_fd(-1)
E       ValueError: set_wakeup_fd only works in main thread

C:\Program Files\Python38\lib\asyncio\proactor_events.py:679: ValueError
-------------------------------------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------------------------------------- 
{'startTime': '1:10', 'endTime': '2:20'}
_______________________________________________________________________________________________________ test_delete_appointment _______________________________________________________________________________________________________ 

    def test_delete_appointment():
>       response = client.delete('/devsecops/vaccine/1')

tests\testvaccine.py:45:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
..\venv\lib\site-packages\requests\sessions.py:669: in delete
    return self.request("DELETE", url, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:473: in request
    return super().request(
..\venv\lib\site-packages\requests\sessions.py:587: in request
    resp = self.send(prep, **send_kwargs)
..\venv\lib\site-packages\requests\sessions.py:701: in send
    r = adapter.send(request, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:267: in send
    raise exc
..\venv\lib\site-packages\starlette\testclient.py:264: in send
    portal.call(self.app, scope, receive, send)
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\starlette\testclient.py:451: in _portal_factory
    yield portal
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\anyio\from_thread.py:502: in start_blocking_portal
    run_future.result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:439: in result
    return self.__get_result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:388: in __get_result
    raise self._exception
C:\Program Files\Python38\lib\concurrent\futures\thread.py:57: in run
    result = self.fn(*self.args, **self.kwargs)
..\venv\lib\site-packages\anyio\_core\_eventloop.py:70: in run
    return asynclib.run(func, *args, **backend_options)
..\venv\lib\site-packages\anyio\_backends\_asyncio.py:292: in run
    return native_run(wrapper(), debug=debug)
C:\Program Files\Python38\lib\asyncio\runners.py:50: in run
    loop.close()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ProactorEventLoop running=False closed=False debug=False>

    def close(self):
        if self.is_running():
            raise RuntimeError("Cannot close a running event loop")
        if self.is_closed():
            return

>       signal.set_wakeup_fd(-1)
E       ValueError: set_wakeup_fd only works in main thread

C:\Program Files\Python38\lib\asyncio\proactor_events.py:679: ValueError
________________________________________________________________________________________________________ test_create_vaccine_1 ________________________________________________________________________________________________________

    def test_create_vaccine_1():
>       response = client.post('/devsecops/vaccine', json=data1)

tests\testvaccine.py:58:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
..\venv\lib\site-packages\requests\sessions.py:635: in post
    return self.request("POST", url, data=data, json=json, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:473: in request
    return super().request(
..\venv\lib\site-packages\requests\sessions.py:587: in request
    resp = self.send(prep, **send_kwargs)
..\venv\lib\site-packages\requests\sessions.py:701: in send
    r = adapter.send(request, **kwargs)
..\venv\lib\site-packages\starlette\testclient.py:267: in send
    raise exc
..\venv\lib\site-packages\starlette\testclient.py:264: in send
    portal.call(self.app, scope, receive, send)
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\starlette\testclient.py:451: in _portal_factory
    yield portal
C:\Program Files\Python38\lib\contextlib.py:120: in __exit__
    next(self.gen)
..\venv\lib\site-packages\anyio\from_thread.py:502: in start_blocking_portal
    run_future.result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:439: in result
    return self.__get_result()
C:\Program Files\Python38\lib\concurrent\futures\_base.py:388: in __get_result
    raise self._exception
C:\Program Files\Python38\lib\concurrent\futures\thread.py:57: in run

>       signal.set_wakeup_fd(-1)
E       ValueError: set_wakeup_fd only works in main thread

C:\Program Files\Python38\lib\asyncio\proactor_events.py:679: ValueError
======================================================================================================= short test summary info ======================================================================================================= 
FAILED tests/testvaccine.py::test_retrieve_appointments_byid - ValueError: set_wakeup_fd only works in main thread
FAILED tests/testvaccine.py::test_create_vaccine - ValueError: set_wakeup_fd only works in main thread
FAILED tests/testvaccine.py::test_retrieve_all_appointments - ValueError: set_wakeup_fd only works in main thread
FAILED tests/testvaccine.py::test_update_appointment - ValueError: set_wakeup_fd only works in main thread
FAILED tests/testvaccine.py::test_delete_appointment - ValueError: set_wakeup_fd only works in main thread
FAILED tests/testvaccine.py::test_create_vaccine_1 - ValueError: set_wakeup_fd only works in main thread
========================================================================================================== 6 failed in 2.72s ========================================================================================================== 
(venv) PS C:\Users\Downloads\FastAPI-main\vaccine>

**Comment by DineshGon** [comment]
On my personal Mac i can able to run the test cases but whereas on my company windows i am getting the above error. Thank you for your response. I will take a look why it causing that issue.

**Comment by tiangolo** [comment]
Thanks for the help here @JarroVGIT ! 👏  🙇 

Thanks for reporting back and closing the issue @fargonmer  👍
